### PR TITLE
improve code

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: frontend/src/App.vue [hunk 1]
```diff
@@ -94,44 +95,95 @@ export default {
...
   watch: {
-    "currentPrompt.answers": {
+    "prompts": {
+      handler(promptsVal) {
+        if (_.isEmpty(promptsVal)) {
+          this.showBusyIndicator = true
+        } else {
+          this.showBusyIndicator = false
+        }
+      }
```
Comment (tomer-epstein) [inline frontend/src/App.vue]: this. setBusyIndicator()
--------------------------------------------------
methods: {
    setBusyIndicator() {
      this.showBusyIndicator = !this.prompts.length || (this.currentPrompt.status === 'pending' && !this.isDone);
    },

File: frontend/src/App.vue [hunk 2]
```diff
@@ -94,44 +95,95 @@ export default {
...
+    },
+    "currentPrompt.status": {
+      handler(statusVal) {
+        if (statusVal === PENDING && !this.isDone) {
+          this.showBusyIndicator = true
+        } else {
+          this.showBusyIndicator = false
+        }
```
Comment (tomer-epstein) [inline frontend/src/App.vue]: setBusyIndicator()

File: frontend/src/App.vue [hunk 3]
```diff
@@ -94,44 +95,95 @@ export default {
...
+      }
+    },
+    "isDone": {
+      handler(isDoneVal) {
+        if (this.currentPrompt.status === PENDING && !isDoneVal) {
+          this.showBusyIndicator = true
+        } else {
+          this.showBusyIndicator = false
+        }
```
Comment (tomer-epstein) [inline frontend/src/App.vue]: setBusyIndicator()

File: frontend/src/App.vue [hunk 4]
```diff
@@ -124,29 +124,17 @@ export default {
   watch: {
     "prompts": {
       handler(promptsVal) {
-        if (_.isEmpty(promptsVal)) {
-          this.showBusyIndicator = true
-        } else {
-          this.showBusyIndicator = false
-        }
+        this.setBusyIndicator(_.isEmpty(promptsVal))
```
Comment (tomer-epstein) [inline frontend/src/App.vue]: I would call this.setBusyIndicator() without params, and let the func to calculate isBussy.
1 flow can set busy to false while other flow is still in busy - true.

File: frontend/src/App.vue (hunk 1 above)
Comment (slavik-lvovsky) [reply frontend/src/App.vue]: Done

**Pull request merged**